src/StakingClient.ts

```diff
@@ -0,0 +1,174 @@
+   */
+   async getContractAddress(params: { contractId: number }): Promise<string> {
+    await this.initialiseReachAccount();
+    const ctc = this.reachAccount.contract(backend, params.contractId);
+    const contractAddress = await ctc.getContractAddress();
+    return this.reachStdLib.formatAddress(contractAddress);
+  }
+
+  /**
+   * 
+   * @param params contractId which indicates the contract we want to interact with
+   */
+  async getContractAbi(params: { contractId: number }): Promise<AbiInterface> {
+    await this.initialiseReachAccount();
+    const ctc = this.reachAccount.contract(backend, params.contractId);
+    return await ctc.getABI();
+  }
+  
+  /**
+   * @description Get global state of contract
+   * @param params account object that contains the reach account
+   * @returns StakeGlobalView
+   */
+  async getState(params: { account: Account }): Promise<StakeGlobalView> {
+    const ctc = params.account.reachAccount.contract(backend, this.id);
+    const get = ctc.v.State;
+    const stateView = await get.read();
+    if (stateView[0] === 'None') {
+      throw new Error('The view is null');
+    }
+    const stakeState = stateView[1];
+    return {
+      stakingAssetID: stakeState.stakingAsset.toNumber(),
+      policy: {
+        applicationType: stakeState.policy.applicationType.toNumber(),
+        status: stakeState.policy.status.toNumber(),
+        time: stakeState.policy.time.toNumber(),
+        rewardMethod: stakeState.policy.rewardMethod.toNumber(),
+        rewardRate: stakeState.policy.rewardRate.toNumber()
+      },
+      totalStake: stakeState.totalStake.toNumber(),
+      totalUsers: stakeState.totalUsers.toNumber(),
+      totalRewards: stakeState.totalRewards.toNumber()
+    };
+  }
+
+  /**
+   * @description Get local state of user
+   * @param params account object that contains the reach account, address for the look up
+   * @returns StakeLocalView
+   */
+  async getUserInfo(params: { account: Account; address: string }): Promise<StakeLocalView> {
+    const ctc = params.account.reachAccount.contract(backend, this.id);
+    const get = ctc.v.State;
+    const stateView = await get.readUserAccount(params.address);
+    if (stateView[1][0] === 'None') {
+      return {
+        amountStaked: stateView.amountStaked.toNumber(),
+        rewardsClaimed: stateView.rewardsClaimed.toNumber(),
+        lastClaim: stateView.lastClaim.toNumber(),
+        found: false
+      };
+    }
+    const stakingState = stateView[1][1];
+    return { ...StakingClient.parseUserInfo(stakingState), found: true };
+  }
+
+  /**
+   * @description stake assets to the contract
+   * @param amount amount to be staked
+   * @returns boolean
+   */
+  async stakeAsset(amount: number): Promise<boolean> {
+    await this.initialiseReachAccount();
+    const ctc = this.reachAccount.contract(backend, this.id);
+    const put = ctc.a.LiquidationVaultAPI;
+    return put.stakeAsset(amount);
```

Comment:
Just for consistency, so far we seem to have used an await to call the api functions and return its result, for example
```js
const res = await put.mintToken(convertToMicroUnits(params.amount));
    return res;
```
should we stick to the above or we should stick to what we have presently here

Reply:
I think in most cases we should `return function()` if we don't need to modify the returned data before returning it to the caller.

Good to call this out - we should lint against this I think

Reply:
For now, I don't think I will change anything in this PR